### ISSUE TITLE
Surround policy ids in backticks to aid readability 

### ISSUE THREAD

**Issue by aaronjeline** [cedar-policy/cedar#727]
### Category

Error message improvements

### Describe the feature you'd like to request

When the validator returns an error, it should quote the policy id with backticks to aid reading. Example:
```
An error occurred Error: Validation error on policy fjdmaoljifeolaj: Policy is impossible. The policy expression evaluates to false for all valid requests
```
should become:
```
An error occurred Error: Validation error on policy `fjdmaoljifeolaj`: Policy is impossible. The policy expression evaluates to false for all valid requests
```

I'm going to suggest we avoid doing "proper quoting" and just don't worry about internal backticks for now, but if people disagree we can hash it out in the comments. 

### Describe alternatives you've considered

Do proper backtick escaping

### Is this something that you'd be interested in working on?

- [ ] 👋 I may be able to implement this feature request
- [ ] ⚠️ This feature might incur a breaking change

**Comment by memark** [comment]
I could take a look at this.

Looking at the code, the specific example already seems solved. This is because the error above is now a `ValidationWarning`, and warnings are already properly formatted.
https://github.com/cedar-policy/cedar/blob/main/cedar-policy-validator/src/validation_result.rs#L355

Should I implement a similar `Display` trait for `ValidationError`?

**Comment by john-h-kastner-aws** [comment]
Yeah, the structs for these errors got reworked in #804, and it looks like the exact error text got changed around a bit.

I think this could be resolved just by adjusting the `error` macro on `Validation` error here https://github.com/cedar-policy/cedar/blob/7708727df28c944906418514291c713de2d6c185/cedar-policy-validator/src/validation_result.rs#L80 and then I might move the `Display` impl for `ValidationWarning` into a similar `error` macro since I don't think there's any reason it needs to manually implement the trait.

Definitely take a look at making the change if you want

**Comment by memark** [comment]
Thanks, good pointers. I'm able to refactor the warning messages without trouble using the macro.

---

As for the error message text it currently looks like this:

example warning:
> validation warning on policy `some_policy`: identifier `some_identifier` contains mixed scripts

example error:
> unrecognized entity type `invalid_type`

Just double-checking that we want the error messages on the longer form as well?

---

Lastly I would like to align the property names of the structs

```
pub struct ValidationWarning {
    pub(crate) location: SourceLocation,
    pub(crate) kind: ValidationWarningKind,
}
```

```
pub struct ValidationError {
    location: SourceLocation,
    error_kind: ValidationErrorKind,
}
```

I would suggest using the shorter name `kind` in both structs.

**Comment by john-h-kastner-aws** [comment]
Aligning on `kind` would be good.

The longer form might be a little too verbose. There's no reason for us to explicitly include `validation error` or `validation warning` on every message. We do want to include policy id since that's important debuging information for policies loaded from JSON where we don't currently include source locations in error messages.

They could be something like ``for policy `{id}`, {msg}``

**Comment by memark** [comment]
Great, I'll do that in two steps, as separate PRs then.

**Comment by memark** [comment]
@aaronjeline This issue is completed now and can be closed.
